src/main/java/org/springframework/data/elasticsearch/core/DefaultResultMapper.java

```diff
@@ -193,7 +154,7 @@ private String buildJSONFromFields(Collection<DocumentField> values) {
 		List<T> list = new ArrayList<>();
 		for (MultiGetItemResponse response : responses.getResponses()) {
 			if (!response.isFailed() && response.getResponse().isExists()) {
-				T result = mapEntity(response.getResponse().getSourceAsString(), clazz);
+				T result = mapResult(response.getResponse(), clazz);
```

Comment:
what if `result`is `null`?

Reply:
How can `result` be null in an response indicated the result exists and it's no failure?

Reply:
ok, I see that in practice it cannot be null; but the method itself is declared as potentially return a null (can it be that I'm working too much with Kotlin?)